src/command_modules/azure-cli-cdn/azure/cli/command_modules/cdn/commands.py

```diff
@@ -0,0 +1,122 @@
+from azure.cli.core.commands.arm import cli_generic_update_command
+from ._client_factory import cf_cdn
+
+
+def custom(sub_path):
+    return'azure.cli.command_modules.cdn.custom#{}'.format(sub_path)
+
+
+def mgmt(sub_path):
+    return 'azure.mgmt.cdn#CdnManagementClient.{}'.format(sub_path)
+
+
+def endpoint(sub_path):
+    return 'azure.mgmt.cdn.operations.endpoints_operations#EndpointsOperations.{}'.format(sub_path)
+
+
+def profile(sub_path):
+    return 'azure.mgmt.cdn.operations.profiles_operations#ProfilesOperations.{}'.format(sub_path)
+
+
+def domain(sub_path):
+    path = 'azure.mgmt.cdn.operations.custom_domains_operations#CustomDomainsOperations.{}'
+    return path.format(sub_path)
+
+
+def origin(sub_path):
+    return 'azure.mgmt.cdn.operations.origins_operations#OriginsOperations.{}'.format(sub_path)
+
+
+def edge(sub_path):
+    return 'azure.mgmt.cdn.operations.edge_nodes_operations#EdgeNodesOperations.{}'.format(sub_path)
+
+
+def command(cmd, method, formatter=custom):
+    cli_command(__name__, cmd, formatter(method), cf_cdn)
+
+
+def endpoint_command(name, method=None, use_custom=False):
+    formatter = custom if use_custom else endpoint
+    command('cdn endpoint {}'.format(name), method or name, formatter)
+
+
+def generic_endpoint_command(name, custom_handler):
+    cli_generic_update_command(__name__,
+                               'cdn endpoint {}'.format(name),
+                               endpoint('get'),
+                               endpoint('update'),
+                               cf_cdn,
+                               setter_arg_name='endpoint_update_properties',
+                               custom_function_op=custom(custom_handler))
+
+
+def generic_profile_command(name, custom_handler):
+    cli_generic_update_command(__name__,
+                               'cdn profile {}'.format(name),
+                               profile('get'),
+                               profile('update'),
+                               cf_cdn,
+                               setter_arg_name='profile_update_properties',
+                               custom_function_op=custom(custom_handler))
+
+
+def profile_command(name, method=None, use_custom=False):
+    formatter = custom if use_custom else profile
+    command('cdn profile {}'.format(name), method or name, formatter)
+
+
+def domain_command(name, method=None, use_custom=False):
+    formatter = custom if use_custom else domain
+    command('cdn custom-domain {}'.format(name), method or name, formatter)
+
+
+def origin_command(name, method=None, use_custom=False):
+    formatter = custom if use_custom else origin
+    command('cdn origin {}'.format(name), method or name, formatter)
+
+
+def edge_command(name, method=None, use_custom=False):
+    formatter = custom if use_custom else profile
+    command('cdn edge-node {}'.format(name), method or name, formatter)
+
+
+cli_command(__name__, 'cdn name-exists', mgmt('check_name_availability'), cf_cdn)
```

Comment:
Almost 90 LOC before the command registrations start seems like a lot of boilerplate when there's only 30 or so lines of actual registration.

I would prefer we stick with one of the consistent styles--either the service adapter or the flat approach.

I favor the flat approach.  It does suffer from copypastitis but it's easy to scan and see all the familiar pieces and parts.